bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/events/ItemEventFactory.java

```diff
@@ -113,10 +139,17 @@ private State getState(String type, String value) {
 
     private String getItemName(String topic) {
         String[] topicElements = getTopicElements(topic);
-        if (topicElements.length != 4)
+        if (topicElements.length < 4)
             throw new IllegalArgumentException("Event creation failed, invalid topic: " + topic);
         return topicElements[2];
     }
+    
+    private String getMemberName(String topic) {
+        String[] topicElements = getTopicElements(topic);
+        if (topicElements.length != 5)
```

Comment:
It is no problem to use "!=" or "<".
But I think at least as the member name is the fifth element, I am also fine with six or more elements.
IMHO the getMemberName does not need to depend on if there has more elements added.
I see that getItemName is using that approach before, but I (personally) would prefer usinng "<".